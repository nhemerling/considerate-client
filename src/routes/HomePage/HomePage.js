import React, { Component } from 'react';
import FriendCard from '../../components/FriendCard/FriendCard';
import FriendApiService from '../../services/friend-api-service';
import ConsiderateContext from '../../context/ConsiderateContext';
import moment from 'moment';
import './HomePage.css';

export default class HomePage extends Component {
  static contextType = ConsiderateContext;

  componentDidMount() {
    FriendApiService.getFriends()
      .then(this.context.setFriendList)
      .catch(this.context.setError);
  }

  compare(a, b) {
    const dateA = a.occasion_date;
    const dateB = b.occasion_date;

    let comparison = 0;
    if (dateA > dateB) {
      comparison = 1;
    } else if (dateA < dateB) {
      comparison = -1;
    }
    return comparison;
  }

  // sorts friends in chronological order of occasions then
  // filters friends to not show past and only show occasions in the next month
  renderFilteredFriends() {
    const { friendList = [] } = this.context;
    const sortedFriends = friendList.sort(this.compare);
    const filteredFriendList = sortedFriends.filter(
      (friend) =>
        moment(friend.occasion_date).month() < moment().month() + 1 &&
        moment() < moment(friend.occasion_date)
    );
    return filteredFriendList.map((friend) => (
      <FriendCard key={friend.id} friend={friend} />
    ));
  }

  render() {
    return (
      <section className="HomePage">
        <h2>Upcoming Occasions</h2>
        <div className="HomePage__friends">{this.renderFilteredFriends()}</div>
      </section>
    );
  }
}
