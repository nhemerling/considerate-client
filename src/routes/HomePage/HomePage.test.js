import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import HomePage from './HomePage';

describe('HomePage component', () => {
  it('renders a HomePage component by default', () => {
    const wrapper = shallow(<HomePage />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
