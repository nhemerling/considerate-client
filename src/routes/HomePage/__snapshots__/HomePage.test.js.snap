// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component renders a HomePage component by default 1`] = `
<section
  className="HomePage"
>
  <h2>
    Upcoming Occasions
  </h2>
  <div
    className="HomePage__friends"
  />
</section>
`;
